transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollChannel.java

```diff
@@ -237,14 +237,14 @@ protected final int doReadBytes(ByteBuf byteBuf) throws Exception {
         return localReadAmount;
     }
 
-    protected final int doWriteBytes(ByteBuf buf) throws Exception {
+    protected final int doWriteBytes(ByteBuf buf, int writeSpinCount) throws Exception {
         int readableBytes = buf.readableBytes();
         int writtenBytes = 0;
         if (buf.hasMemoryAddress()) {
             long memoryAddress = buf.memoryAddress();
             int readerIndex = buf.readerIndex();
             int writerIndex = buf.writerIndex();
-            for (;;) {
+            for (int i = writeSpinCount; i >= 0; i--) {
```

Comment:
The initial value has to be `writeSpinCount - 1`. Please see `NioSocketChannel.doWrite()`.